server/src/test/java/org/opensearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -257,24 +258,48 @@ public void testUsesGlobalOrdinalsByDefault() throws Exception {
         directory.close();
     }
 
-    public void testSimple() throws Exception {
+    /**
+     * This test case utilizes the low cardinality implementation of GlobalOrdinalsStringTermsAggregator.
+     * In this case, the segment terms will not get initialized and will run without LeafBucketCollector#termDocFreqCollector optimization
+     */
+    public void testSimpleAggregation() throws Exception {
+        testSimple(ADD_SORTED_FIELD_NO_STORE, DEFAULT_POST_COLLECTION);
+    }
+
+    /**
+     * This test case utilizes the low cardinality implementation of GlobalOrdinalsStringTermsAggregator.
+     * In this case, the segment terms will get initialized and will use LeafBucketCollector#termDocFreqCollector optimization
+     */
+    public void testSimpleAggregationWithStoredValues() throws Exception {
+        // aggregator.postCollection() is not required when LeafBucketCollector#termDocFreqCollector optimization is used,
```

Comment:
postCollection is not required, but it will still be executed. I am not sure the reason of explicitly using `NOOP_POST_COLLECTION`. Without using `NOOP_POST_COLLECTION`, will this test fail?

Reply:
Agreed -- if the aggregator fails on `postCollection` with the optimization we need to fix it.

Maybe we need to store some state in the aggregator when we create the `LeafCollector` to remember that we did the optimization?

Reply:
`NOOP_POST_COLLECTION` just signifies that we are not doing any collection on leaf collectors, still getting the correct results. I have mentioned this note when declaring this variable.